- Added `.gitattributes`, LF End-of-Lines only

KevinTran000 opened the pull request:
\- Added `.gitattributes` that is set for numerous types to be restricted to LF EOLs only.

This resolves issue #82.

Added .gitattribute template from https://github.com/alexkaratarakis/gitattributes/blob/master/Web.gitattributes with minor modification.

Tested and confirms it works as shown in screenshot below:
![2019-04-15_22-14-19](https://user-images.githubusercontent.com/47541021/56177467-b966cd00-5fcc-11e9-8f96-6526f606bd1d.png)

KevinTran000 committed:
- Added `.gitattributes` that is set for numerous types to be restricted to LF EOLs only.

StoneyJackson commented:
@KevinTran000 Thanks for picking this up! Awesome start!

I'm seeing some interesting suggestions here: https://stackoverflow.com/questions/170961/whats-the-best-crlf-carriage-return-line-feed-handling-strategy-with-git

Rather than make one from scratch, I wonder if we should use one of the templates from https://github.com/alexkaratarakis/gitattributes ? Specifically, I'm thinking Web.gitattributes might work well.

The same post also points to a GitHub help page that give instructions for renormalizing the end-of-lines in the project after adding/updating .gitattributes. We should probably do that too.

KevinTran000 committed:
- Modified `.gitattribute` to use template

This is better than commit `5a68587cc` (the first iteration of this file) which covers more file types including the ones from the first iteration.

KevinTran000 commented:
Agreed, the `Web.gitattributes` template would work well and it covers the file types listed in the first iteration and more. Just pushed up a commit with your suggestion implemented.

KevinTran000 committed:
- Added license file for `.gitattributes`

StoneyJackson committed:
Add license to the top of .gitattributes

This way we won't have an extra, potentially confusing, license file hanging around. Also, if we ever get rid of that file, the license will go with it.

StoneyJackson committed:
Move license to .gitattributes file

StoneyJackson committed:
Prevent normalizing EOLs on select .gitignore

The problem is that the MacOS file Icon\r\r has two carriage returns at the end of the file name. If git normalizes them we lose these characters. This commit adds a rule to .gitattributes to not normalize EOL in /.gitignore (which is generated anyway) and /.gitigonre.d/macOS.gitignore (which is where the Icon\r\r rule appears).

aidanjones542 commented:
I approve. We should be enforcing EOL consistency.
